web/src/app/users/UserCreateDialog.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState } from 'react'
+import { gql, useMutation } from '@apollo/client'
+import { Redirect } from 'react-router'
+import { fieldErrors, nonFieldErrors } from '../util/errutil'
+import FormDialog from '../dialogs/FormDialog'
+import { FormContainer, FormField } from '../forms'
+import { Checkbox, FormControlLabel, Grid, TextField } from '@material-ui/core'
+import { useConfigValue } from '../util/RequireConfig'
+
+const mutation = gql`
+  mutation ($input: CreateUserInput!) {
+    createUser(input: $input) {
+      id
+    }
+  }
+`
+const initialValue = {
+  username: '',
+  password: '',
+  email: '',
+  isAdmin: false,
+}
+
+interface UserCreateDialogProps {
+  onClose: () => void
+}
+
+function UserCreateDialog(props: UserCreateDialogProps): JSX.Element {
+  const [value, setValue] = useState(initialValue)
+  const [authDisableBasic] = useConfigValue('Auth.DisableBasic')
+  const [createUser, { loading, data, error }] = useMutation(mutation, {
+    variables: {
+      input: {
+        username: value.username,
+        password: value.password,
+        email: value.email,
+        role: value.isAdmin ? 'admin' : 'user',
+      },
+    },
+  })
+
+  if (data?.createUser) {
+    return <Redirect push to={`/users/${data.createUser.id}`} />
+  }
+
+  return (
+    <FormDialog
+      title='Create User'
+      loading={loading}
+      errors={nonFieldErrors(error)}
+      onClose={props.onClose}
+      onSubmit={() => createUser()}
+      notices={
+        authDisableBasic
+          ? [
+              {
+                type: 'WARNING',
+                message: 'Basic Auth is Disabled',
+                details:
+                  'This user will be unable to log in until basic auth is enabled.',
+              },
+            ]
+          : []
+      }
+      form={
+        <FormContainer
+          value={value}
+          errors={fieldErrors(error)}
+          onChange={(val: typeof value) => setValue(val)}
+        >
+          <Grid container spacing={2}>
+            <Grid item xs={12}>
+              <FormField
+                fullWidth
+                component={TextField}
+                label='Username'
+                name='username'
+                required
+              />
```

Comment:
Username error: `Can only contain lower-case letters and digits`

We should probably allow spaces, as full names are valid users in GoAlert

Reply:
`username` (auth) isn't the same as the user's name/display name. In this case, it's referring to basic auth (i.e. user/pass) specifically and not the display name when that error comes back.

It's set that way to remove ambiguity and things like 2 usernames that appear the same due to whitespace differences, etc.. Name or Display Name should probably be its own field. The command line just set's the display name to be the login username as a convenience, but they are stored and validated separately. `username`, which is typed in the login form, is just stricter